### PR TITLE
feat: install proxy on off1 and serve OFF images

### PR DESCRIPTION
**Pull request by stephanegigandet**
Install log on what I did to install a proxy container (100) on off1 with nginx to serve images from ZFS if we have them locally, or from the off container on off2 otherwise.

I tested it and it works, but there are some TODOs left:

- Also do the setup for OBF, OPF, OPFF: nginx configuration, SSL certificates etc.
- Check why we don't have /zfs-hdd/opf on off1
- Verify that off1 can renew the images.openfoodfacts.org SSL certificate
- Export proxy logs and static logs to prometheus?

**IMPORTANT** this was abandoned immediately due to a performance issue (not yet investigated).

### CODE REVIEW

File: docs/reports/2024-06-05-off1-reverse-proxy-install.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+                  '$status $body_bytes_sent "$http_referer" '
+                  '"$http_user_agent" "$http_x_forwarded_for"';
+```
+It's in the git repository, so then:
+
+```bash
+ln -s /opt/openfoodfacts-infrastructure/confs/proxy-off/nginx/log_format.conf /etc/nginx/conf.d/log_format.conf
+```
+
+From off2 static-off configuration, I create a similar images-off site, and put it into git.
```
Comment (CharlesNepote) [inline docs/reports/2024-06-05-off1-reverse-proxy-install.md:120]:
```suggestion
From off2 static-off configuration, I create a similar images-off site, and [put it into git](../confs/off1-reverse-proxy/nginx/sites-available/images-off).
```

**Review by CharlesNepote — APPROVED** [review]
Great.

File: docs/reports/2024-06-05-off1-reverse-proxy-install.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+Failed to renew certificate images.openfoodfacts.org with error: The requested nginx plugin does not appear to be installed
+
+Installed python3-certbot-nginx
+
+2nd try:
+
+Detail: 213.36.253.208: Fetching http://images.openfoodfacts.org/.well-known/acme-challenge/GUG0men38f_steRovK_AmCPyaV4lCyJUA3N8teSWp3I: Connection refused
+
+TODO: check if this is because the live images.openfoodfacts.org currently points to off2.
```
Comment (alexgarel) [inline docs/reports/2024-06-05-off1-reverse-proxy-install.md:103]: Yes, you can't renew until you have images.openfoodfacts.org pointing to your new IP address.

File: docs/reports/2024-06-05-off1-reverse-proxy-install.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+We reuse the existing `confs/proxy-off/fail2ban/jail.d/nginx` using debian provided feature for nginx
+Then:
+```bash
+sudo ln -s /opt/openfoodfacts-infrastructure/confs/proxy-off/fail2ban/jail.d/nginx.conf /etc/fail2ban/jail.d/
+systemctl reload fail2ban
+```
+
+**NOTE**: it's really not enough, but to analyze 403 / 401 we need a specific plugin that analyze logs.
+
```
Comment (alexgarel) [inline docs/reports/2024-06-05-off1-reverse-proxy-install.md:66]: You can eventually also do the setup to have the manual ban jail (to be able to ban bots)

**Review by alexgarel — APPROVED** [review]
Great. Thank you for documenting precisely.

**Comment by stephanegigandet** [comment]
missing /zfs-hdd/opf on off1 issue fixed by https://github.com/openfoodfacts/openfoodfacts-infrastructure/pull/355

**Comment by alexgarel** [comment]
Tested ok on my computer, changing /etc/hosts !

**Pull request merged**